### PR TITLE
Express 102: CRUD and MVC: Address database credential security risk

### PR DESCRIPTION
**Pull request by EmilyErickson**
## Because

There was a security risk in the Express & Mongoose section of the NodeJS course.
The MDN library tutorial instructions has students hardcoding their database credentials. There is a brief warning in the last MDN lesson to hide database usernames and passwords, but there isn't a clear explanation on how to do that. An explanation of how to hide sensitive information and avoid pushing database credentials to Github was needed.

## This PR

- Adds an assignment to the Express 102: CRUD and MVC lesson that has students read about using environment variables and instructs them to not hardcode their database credentials.
- Re-words the reminder in the Project: Express 105: Forms and Deployment lesson  to verify that credentials are safely hidden.

## Issue
Closes [#27415](https://github.com/TheOdinProject/curriculum/issues/27415)

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by EmilyErickson** [comment]
I am a little confused as to why some of the checks failed. Some of the errors that I am getting are things that I didn't change at all. I would be happy to try to fix the errors, but I am not sure if that is necessary as the errors seem irrelevant.

**Comment by cakegod** [comment]
We use a markdown linter to enforce the layout styles. You check those issues locally using the package or the plugin as mentioned in the [CONTRIBUTING file](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md#curriculum-linting).

**Comment by EmilyErickson** [comment]
I am unsure about some of the MarkdownLint errors. For example, the assignment throws this error "MD029/ol-prefix: Ordered list item prefix [Expected: 1; Actual: 2; Style: 1/1/1]". The error goes away when I change the second assignment to be prefixed with 1. but then it doesn't match the syntax of other lessons.
Am I missing something with how to use MarkdownLint?

**Comment by cakegod** [comment]
> then it doesn't match the syntax of other lessons.

Mind elaborating on what you mean?

That style guide on the specific rule mentions that you should use "lazy numbering". By using `1.` on every list item, the number is automatically and sequentially incremented.

**Comment by EmilyErickson** [comment]
I didn't realize it would automatically increment the numbers. That fixed my problem. Thanks so much!

**Comment by EmilyErickson** [comment]
I keep getting an error about the heading syntax. For example, I get this error "nodeJS/express_and_mongoose/express_102_crud_and_mvc.md:73 TOP004/lesson-headings Required heading structure [Expected: h4 heading; Actual: h3 heading]" but when I change the "### Additional resources" heading to "#### Additional resources" I am getting this error "nodeJS/express_and_mongoose/express_102_crud_and_mvc.md:79 TOP004/lesson-headings Required heading structure [Context: "### Knowledge check"]"
I am not sure how to fix it and I haven't been able to find any rules explaining it.

File: nodeJS/express_and_mongoose/express_102_crud_and_mvc.md [hunk 1]
```diff
@@ -19,58 +19,60 @@ This section contains a general overview of topics that you will learn in this l
...
 
 <div class="lesson-content__panel" markdown="1">
 
+1. If you're planning to publish your code on GitHub you will need to hide the MongoDB connection URL beacuse it contains your username and password. One way to do this is to push your code to a private repository. However, if you want to show-off your code, you can assign your connection url to an environment variable and access it with [the Dotenv package](https://github.com/motdotla/dotenv). This[Medium article](https://javascript.plainenglish.io/how-to-use-dotenv-and-why-you-need-it-in-all-your-projects-a7319e16124a) does a great job of explaining what environment variables are and how to use them with Express. Read through the article and then when you reach the section that has you connect to MongoDB, follow the instructions in this article to hide your connection string and access it from the `.env` file.
+
 1. Complete part 3 of [the MDN Express tutorial](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/mongoose).
 
 </div>
 
 ### Knowledge checks
```
Comment (cakegod) [inline nodeJS/express_and_mongoose/express_102_crud_and_mvc.md:64]:
```suggestion
### Knowledge check
```
This is what is causing the error. The custom rule error is not very clear.

Comment (EmilyErickson) [reply nodeJS/express_and_mongoose/express_102_crud_and_mvc.md]: I missed that typo and some of the rules are a little confusing. Thank you so much for your help!

**Comment by EmilyErickson** [comment]
@bycdiaz - Just checking to see if there is anything else that needs with this done before it can be reviewed/merged.
Thanks!

**Comment by cakegod** [comment]
I recommend you to request a new review. See the step five at the bottom of this [article about requesting a pull review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review). You should do this everytime you apply or fix what was suggested in the review.

**Comment by bycdiaz** [comment]
@EmilyErickson Looks like there is a merge conflict. Let's get that taken care of.

**Comment by MaoShizhong** [comment]
Closing this as the merge of milestone 1 of the Node revamp addresses this
(https://github.com/TheOdinProject/theodinproject/pull/4582#event-13561502963)

**Pull request closed without merging**